### ISSUE TITLE
[FEA] DataFrame correlation

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#3675]
Generating the correlation matrix of a dataframe is fairly time-consuming for tall and wide dataframes in pandas. We should support DataFrame level correlation with #1267 eventually, but it would be nice to support it in Python via CuPy as a stopgap. This would help our compatibility for exploratory analysis (some of which is commonly automated via pandas-ecosystem libraries like [pandas-profiling](https://github.com/pandas-profiling/pandas-profiling/)).

We may want to roll this into #1267 , in which case we can close this issue. Follow-up to #2719 (see discussion). This can be implemented just like `DataFrame.cov` in #3393

**Comment by beckernick** [comment]
```python
    return df
import cudf
from cudf import DataFrame
import cupy
​
def frame_corr(self):
    cov = cupy.corrcoef(self.values, rowvar=False)
    df = DataFrame.from_gpu_matrix(cupy.asfortranarray(cov)).set_index(
        self.columns
    )
    df.columns = self.columns
    return df
​
# 1 million x 50
df = cudf.datasets.randomdata(1000000, {str(x):float for x in range(50)})
pdf = df.to_pandas()
​
%time corr = pdf.corr()
%time corrgpu = frame_corr(df)
CPU times: user 4.44 s, sys: 164 ms, total: 4.6 s
Wall time: 4.6 s
CPU times: user 46.2 ms, sys: 10.8 ms, total: 57 ms
Wall time: 58.5 ms
# takes 1 second when not already "hot", so still fast.
```

**Comment by rnyak** [comment]
I am getting `AttributeError: 'DataFrame' object has no attribute 'corr'` when I run `gdf.corr()`. I am using 0.12 release in docker container. 

Is that still an open issue, or was it already resolved for 0.12 and/or 0.13?

**Comment by beckernick** [comment]
@rnyak this is the open issue for dataframe correlation. It has not yet been implemented in libcuDF. 

Would you like to contribute the temporary CuPy-based approach linked in the comment above in a pull request?

**Comment by rnyak** [comment]
@beckernick Thanks for quick response. Yes, sure, will implement the CuPy-based approach. I am trying to do `corr()` between certain columns  after `groupby`. Hope the proposed approach would work for that.

**Comment by beckernick** [comment]
Updating this issue to explicitly refer to a libcuDF implementation now that a Python implementation has landed #4140

**Comment by tyoc213** [comment]
escuse me, how I can make something like this (which works on pandas)

```python
cu_df = cudf.read_csv("titanic.csv", header=0)
# Correlation matrix
plt.matshow(cu_df.corr())
continuous_features = cu_df.describe().columns
plt.xticks(range(len(continuous_features)), continuous_features, rotation='45')
plt.yticks(range(len(continuous_features)), continuous_features, rotation='45')
plt.colorbar()
plt.show()
```
